inst/include/xml2_types.h

```diff
@@ -5,6 +5,24 @@
 #define R_NO_REMAP
 #include <Rinternals.h>
 
+enum NodeType {
+  missing = 1,
+  node = 2,
+  nodeset = 3,
+};
+
+inline const NodeType getNodeType(SEXP x) {
+  if (Rf_inherits(x, "xml_node")) {
+    return(NodeType::node);
+  } else if (Rf_inherits(x, "xml_nodeset")) {
+    return(NodeType::nodeset);
+  } else if (Rf_inherits(x, "xml_missing")) {
+    return(NodeType::missing);
+  } else {
+    Rf_error("Unexpected node type");
+  }
+}
+
```

Comment:
Is there a reason you put this here instead of, say, `xml2_utils.h` in `src/`?

If you put it here, it means any package that links against xml2 would also see this helper, and I don't think we want to make it public.

(for example, xslt by Jeroen https://github.com/ropensci/xslt/blob/56ceeeaee604fa872bb3f161d52a6b4b6a82796d/src/xslt.cpp#L12)

Reply:
Ah, good point. I simply put it there because the file names seemed to suit well. But I didn't consider that this would be public then.